lib/internal/process/stdio.js

```diff
@@ -8,26 +8,64 @@ function setupStdio() {
   function getStdout() {
     if (stdout) return stdout;
     stdout = createWritableStdioStream(1);
+
+    Object.defineProperty(stdout._writableState, 'ended', {
+      set() {},
+      get() {
+        return false;
+      }
+    });
+    // XXX(Fishrock123): This should work I think, but it causes
+    // problems with some child-process related tests.
+    //
+    // Object.defineProperty(stdout, 'destroyed', {
+    //   set() {},
+    //   get() {
+    //     return true;
+    //   }
+    // });
+
     stdout.destroy = stdout.destroySoon = function(er) {
+      stdout.destroyed = true;
       er = er || new Error('process.stdout cannot be closed.');
       stdout.emit('error', er);
     };
     if (stdout.isTTY) {
       process.on('SIGWINCH', () => stdout._refreshSize());
     }
+    stdout.on('error', () => {});
```

Comment:
I'm not convinced that we should do this in the first place. I'm with @addaleax.

Even if we land, a comment on _why_ that handler is there is 100% needed.